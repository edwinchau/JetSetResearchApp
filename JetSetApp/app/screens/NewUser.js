import React, { Component } from 'react';
import {Text, View, Image, TouchableOpacity} from 'react-native';

import styles from '../resources/Styles'
import {Notifications} from "expo";

export default class NewUser extends Component {
    static navigationOptions = () => {
        return {
            headerStyle: styles.headerStyle,
            headerTintColor: 'black',
            headerTitle:"Home Page",
            headerTitleStyle: {
                flex: 1,
                textAlign: 'center',
            }
        };
    };

    scheduleNotification = async () => {
        let breakfastNotification = await Notifications.scheduleLocalNotificationAsync(
            {
                title: "Breakfast survey and PVT reminder (1hr)",
                body: "Don't forget to fill in your breakfast survey and do the PVT test."
            },
            {
                repeat: "day",
                // 1 hr
                time: new Date().getTime() + 3600
            }
        );
        let lunchNotification = await Notifications.scheduleLocalNotificationAsync(
            {
                title: "Lunch survey and PVT reminder (7hrs)",
                body: "Don't forget to fill in your lunch survey and do the PVT test."
            },
            {
                repeat: "day",
                // 7 hrs
                time: new Date().getTime() + 25200
            }
        );
        let dinnerNotification = await Notifications.scheduleLocalNotificationAsync(
            {
                title: "Dinner survey and PVT reminder (13hrs)",
                body: "Don't forget to fill in your dinner survey and do the PVT test."
            },
            {
                repeat: "day",
                // 13 hours
                time: new Date().getTime() + 46800
            }
        );
    };

    scheduleFlightNotification = async () => {
        let firstMealNotification = await Notifications.scheduleLocalNotificationAsync(
            {
                title: "ON FLIGHT: First meal survey reminder",
                body: "Don't forget to record when you ate your first meal!"
            },
            {
                // 1 hour since departure
                time: new Date().getTime() + 3600
            }
        );
        let secondMealNotification = await Notifications.scheduleLocalNotificationAsync(
            {
                title: "ON FLIGHT: Second meal survey reminder",
                body: "Don't forget to record when you ate your second meal!"
            },
            {
                // 5 hours since departure
                time: new Date().getTime() + 18000
            }
        );
        let thirdMealNotification = await Notifications.scheduleLocalNotificationAsync(
            {
                title: "ON FLIGHT: Second meal survey reminder",
                body: "Don't forget to record when you ate your second meal!"
            },
            {
                // 16 hours since departure
                time: new Date().getTime() + 57600
            }
        );
    };

    render() {
        const { navigation } = this.props;

        return (
            <View style={styles.background}>
                <View style={styles.elementPadding}>
                    <Image style={styles.usydlogo} source={require('../../assets/usydlogo.png')}/>
                </View>
                <View style={styles.elementPadding}>
                    <Text style={styles.mainText}>JetSet Survey App</Text>
                </View>

                <View style={styles.elementPadding}>
                    <TouchableOpacity
                        style={styles.button}
                        onPress={() => {
                            this.scheduleNotification();
                            navigation.navigate('Survey', { survey: 'NewUserQuestions' });
                        }}
                    >
                        <Text style={styles.buttonText}> New User </Text>
                    </TouchableOpacity>
                </View>
            </View>
        );
    }
}
