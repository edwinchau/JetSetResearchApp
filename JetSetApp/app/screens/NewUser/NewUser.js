import React, { Component } from 'react';
import {Text, View, Image, TouchableOpacity} from 'react-native';

import styles from '../../resources/Styles'

export default class NewUser extends Component {
    static navigationOptions = () => {
        return {
            headerStyle: styles.headerStyle,
            headerTintColor: 'black',
            headerTitle:"Home Page",
            headerTitleStyle: {
                flex: 1,
                textAlign: 'center'
            }
        };
    };

    render() {
        const { navigation } = this.props;

        return (
            <View style={styles.background}>
                <View style={styles.elementPadding}>
                    <Image style={styles.usydlogo} source={require('../../../assets/usydlogo.png')}/>
                </View>

                <View style={styles.elementPadding}>
                    <Text style={{fontWeight: "bold", fontSize: 20}}>JetSet Survey App</Text>
                </View>

                <View style={styles.elementPadding}>
                    <TouchableOpacity
                        style={styles.adminButton}
                        onPress={() => navigation.navigate('Survey', { survey: 'NewUserQuestions' })}
                    >
                        <Text style={styles.buttonTextColour}> New Users </Text>
                    </TouchableOpacity>
                </View>
            </View>
        );
    }
}
