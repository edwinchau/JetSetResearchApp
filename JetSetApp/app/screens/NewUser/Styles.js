import { StyleSheet } from 'react-native';

let styles = StyleSheet.create({
    background: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'rgba(244,204,112,1)',
    },
    container: {
        minWidth: '70%',
        maxWidth: '90%',
        alignItems: 'stretch',
        justifyContent: 'center',
        backgroundColor: 'white',
        elevation: 20,
        borderRadius: 15,
        paddingTop: 13,
        paddingBottom: 13,
        marginTop: 10,
        marginBottom: 10,
    },
    questionText: {
        marginBottom: 20,
        fontSize: 20
    },
    headerStyle: {
        backgroundColor: 'rgba(106,177,135,1)',
        height: 40,
        elevation: 5,
    },
});

module.exports = styles;
