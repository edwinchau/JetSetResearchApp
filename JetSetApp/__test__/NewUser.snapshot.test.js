import React from "react";
import NewUser from "../app/screens/NewUser/NewUser"
import renderer from "react-test-renderer"

describe("<NewUser/>", ()=>{
  it("snapshot", ()=>{
    expect(renderer.create(<NewUser/>).toJSON()).toMatchSnapshot();
  });
});
