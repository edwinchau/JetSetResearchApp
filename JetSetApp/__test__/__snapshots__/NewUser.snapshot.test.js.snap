// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewUser/> snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#DF4425",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "paddingVertical": 5,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      JetSet Survey App
    </Text>
  </View>
  <View
    style={
      Object {
        "paddingVertical": 5,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "black",
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
         New User 
      </Text>
    </View>
  </View>
</View>
`;
